asapdiscovery-simulation/asapdiscovery/simulation/schema/fec.py

```diff
@@ -0,0 +1,387 @@
+from gufe import settings
+from openfe.protocols.openmm_rfe.equil_rfe_settings import (
+    AlchemicalSamplerSettings,
+    AlchemicalSettings,
+    IntegratorSettings,
+    OpenMMEngineSettings,
+    SimulationSettings,
+    SolvationSettings,
+    SystemSettings,
+)
+from openff.models.types import FloatQuantity
+from openff.units import unit as OFFUnit
+from pydantic import BaseSettings, Field
+
+
+class AlchemiscaleSettings(BaseSettings):
+    """
+    General settings class to capture Alchemiscale credentials from the environment.
+    """
+
+    ALCHEMISCALE_ID: str = Field(
+        ..., description="Your personal alchemiscale ID used to login."
+    )
+    ALCHEMISCALE_KEY: str = Field(
+        ..., description="Your personal alchemiscale Key used to login."
+    )
+
+
+class SolventSettings(_SchemaBase):
+    """
+    A settings class to encode the solvent used in the OpenFE FEC calculations.
+    """
+
+    type: Literal["SolventSettings"] = "SolventSettings"
+
+    smiles: str = Field("O", description="The smiles pattern of the solvent.")
+    positive_ion: str = Field(
+        "Na+",
+        description="The positive monoatomic ion which should be used to neutralize the system and to adjust the ionic concentration.",
+    )
+    negative_ion: str = Field(
+        "Cl-",
+        description="The negative monoatomic ion which should be used to neutralize the system and to adjust the ionic concentration.",
+    )
+    neutralize: bool = Field(
+        True,
+        description="If the net charge of the chemical system should be neutralized by the ions defined by `positive_ion` and `negative_ion`.",
+    )
+    ion_concentration: FloatQuantity["molar"] = Field(  # noqa: F821
+        0.15 * OFFUnit.molar,
+        description="The ionic concentration required in molar units.",
+    )
+
+    def to_solvent_component(self) -> gufe.SolventComponent:
+        return gufe.SolventComponent(**self.dict(exclude={"type"}))
+
+
+# TODO make base class with abstract methods to collect results.
+class TransformationResult(_SchemaBaseFrozen):
+    """
+    Store the results of a transformation, note when retries are used this will be the average result.
+    """
+
+    type: Literal["TransformationResult"] = "TransformationResult"
+    ligand_a: str = Field(
+        ..., description="The name of the ligand in state A of the transformation."
+    )
+    ligand_b: str = Field(
+        ..., description="The name of the ligand in state B of the transformation."
+    )
+    phase: Literal["complex", "solvent"] = Field(
+        ..., description="The phase of the transformation."
+    )
+    estimate: FloatQuantity["kcal/mol"] = Field(  # noqa: F821
+        ..., description="The average estimate of this transformation in kcal/mol"
+    )
+    uncertainty: FloatQuantity["kcal/mol"] = Field(  # noqa: F821
+        ...,
+        description="The standard deviation of the estimates of this transform in kcal/mol",
```

Comment:
It is not generally the case that the `estimate` is an average. It is the case for the `openfe` `RelativeHybridTopologyProtocol`, where more repeats means more resolved distribution of the samples, with `estimate` as mean and `uncertainty` as standard deviation.

But it isn't the case for the `perses` `NonEquilibriumCyclingProtocol`. More repeats means more accumulated nonequilibrium work values, which are aggregated together to give a single, hopefully *more converged*, estimate. Uncertainty is calculated via bootstrap sampling of the accumulated nonequilibrium work values to quantify the distribution of possible estimates.

The most general way to document `estimate` and `uncertainty` here are that these are the "best" estimates for both the free energy difference and its uncertainty for the `Transformation` given the data.

Reply:
Followup issue #464